### DISCUSSION TITLE
Is Cassowary slowing system shutdown ?

### DISCUSSION THREAD

**Discussion by KaKi87** [casualsnek/cassowary#124] [Q&A] [answered]
Hello,

I had my splash screen disabled by accident and saw the following : 

![](https://github.com/casualsnek/cassowary/assets/21284089/c21115af-e750-46de-b41a-9b0d30893261)

It's been a long time since I noticed how much time my system takes to shut down, yet I thought it was a power management thing like I had on previous systems.

But this log mentions *libvirt* so I'm supposing it's due to Cassowary.

Am I right ? How to fix it ?

Thanks

**Comment by casualsnek** [comment]
Seems to be libvirt witing for vm (that was started by cassowary) to shutdown ! What distribution are you on ?

**Reply by KaKi87** [reply]
Linux Mint

**Reply by KaKi87** [reply]
Hello,
Any news on this ?
Thanks

**Reply by casualsnek** [reply, marked as answer]
Sorry for being late !
The issue from what i guess is due to some windows application being action and since windows presents a confirmation dialog when application are running which cannot be confirmed when host is shutting own causing the script to wait for too long !

Download and run the linked .reg file in windows VM to disable this behaviour which should fix this issue : https://casualsnek.eu.org/shared/DISABLE_END_DIALOG_ON_ACPI_POWEROFF.reg

If that does not work, edit SHUTDOWN_TIMEOUT=300 to something smaller like SHUTDOWN_TIMEOUT=10 at /usr/lib/libvirt/libvirt-guests.sh. This should allow for a quicker shutdown !

**Reply by KaKi87** [reply]
The first solution worked, thanks !

**Reply by KaKi87** [reply]
Update : turns out the first solution only works when Cassowary hasn't been used. But, when at least one Windows app has been used, the issue still occurs, so I had to use the second solution as well. However, I also noticed this change gets reverted during system updates.

Thanks

**Reply by KaKi87** [reply]
Hi, it would be nice to have a permanent fix rather than a temporary workaround on this please, becaude it resets really often. Thanks

**Comment by casualsnek** [comment]
> Hi, it would be nice to have a permanent fix rather than a temporary workaround on this please, becaude it resets really often. Thanks

Thanks for feedback ! I'll see if i can i can fix this issue in a reliable way ! Changing libvirt config will impact other VMs that are not related to cassowary, which may cause unexpected issues. So, maybe i can setup cassowary itself to trigger a guest shutdown when host is shutting down !
